var/spack/repos/builtin/packages/kassiopeia/package.py

```diff
@@ -47,21 +50,12 @@ class Kassiopeia(CMakePackage):
     depends_on('opencl', when='+opencl')
 
     def cmake_args(self):
-        args = []
-        if self.spec.satisfies('+vtk'):
-            args.append('-DKASPER_USE_VTK=ON')
-        else:
-            args.append('-DKASPER_USE_VTK=OFF')
-        if self.spec.satisfies('+tbb'):
-            args.append('-DKASPER_USE_TBB=ON')
-        else:
-            args.append('-DKASPER_USE_TBB=OFF')
-        if self.spec.satisfies('+mpi'):
-            args.append('-DKEMField_USE_MPI=ON')
-        else:
-            args.append('-DKEMField_USE_MPI=OFF')
-        if self.spec.satisfies('+opencl'):
-            args.append('-DKEMField_USE_OPENCL=ON')
-        else:
-            args.append('-DKEMField_USE_OPENCL=OFF')
+        args = [
+            self.define_from_variant("KASPER_USE_VTK", "vtk"),
+            self.define_from_variant("KASPER_USE_TBB", "tbb"),
+            self.define_from_variant("KEMField_USE_MPI", "mpi"),
+            self.define_from_variant("KEMField_USE_OPENCL", "opencl"),
+        ]
+        cxxstd = self.spec['root'].variants['cxxstd'].value
```

Comment:
`root` seems to be a conditional variant, so this line will fail if we try to compile `kassiopea~root`

Reply:
Changed this to use the root cxxstd when +root in spec, otherwise use 14 (default in CMakeLists.txt.